docs/sources/concepts/configuration-syntax/_index.md

```diff
@@ -8,51 +8,81 @@ weight: 10
 
 # {{% param "PRODUCT_NAME" %}} configuration syntax
 
-{{< param "FULL_PRODUCT_NAME" >}} dynamically configures and connects components with the {{< param "PRODUCT_NAME" >}} configuration syntax.
+{{< param "FULL_PRODUCT_NAME" >}} dynamically configures and connects components with the {{< param "PRODUCT_NAME" >}} configuration syntax. Overall, {{< param "FULL_PRODUCT_NAME" >}} handles the collection, transformation, and delivery of telemetry data, and so each component in the configuration handles one of those tasks, or specifies how data flows (for example from collection to transformation), and so how the components are bound together.
 
-The {{< param "PRODUCT_NAME" >}} syntax aims to reduce errors in configuration files by making configurations easier to read and write.
-{{< param "PRODUCT_NAME" >}} configurations use blocks that can be easily copied and pasted from the documentation to help you get started as quickly as possible.
+{{< figure src="/media/docs/alloy/flow-diagram-small-alloy.png" alt="Alloy flow diagram" >}}
 
-An {{< param "PRODUCT_NAME" >}} configuration file tells {{< param "PRODUCT_NAME" >}} which components to launch and how to bind them together into a pipeline.
-
-The {{< param "PRODUCT_NAME" >}} syntax uses blocks, attributes, and expressions.
+Here is a simple example of a fully worked configuration, showing the basic concepts, and how an Alloy config comes together into a pipeline.
 
 ```alloy
-// Create a local.file component labeled my_file.
-// This can be referenced by other components as local.file.my_file.
-local.file "my_file" {
-  filename = "/tmp/my-file.txt"
+// Collection: mount a local directory with a certain path spec
+local.file_match "applogs" {
+    path_targets = [{"__path__" = "/tmp/app-logs/app.log"}]
+}
```

Comment:
`local.file_match` is a weird one, as it fits into the same bucket as the `discovery` components. It's not collecting any data, but performs discovery to enrich other components.

Do you think this distinction is useful at the introduction level? If so, maybe it could be a fourth category of components so the categories become:

1. Data enrichment (local.file_match, discovery.kubernetes, etc.) 
2. Collection 
3. Transformation
4. Delivery 

I'm ok with either; I'll defer to your judgment for what makes the most sense for high-level introductory content.

Reply:
I borrowed this from a worked example my team did, I don't have anything wrapped up in its specifics.

If we can keep the example technically correct while making it shorter & simpler, that's a win.  We specifically don't need to teach anything advanced here, it's only there to demonstrate the "pipeline/flow" concept with a simple example anyone can follow (report logs)